### ISSUE TITLE
Fixed wing mission sometimes result in a flyaway

### ISSUE THREAD

**Issue by Jaeyoung-Lim** [ThunderFly-aerospace/PX4-FlightGear-Bridge#10]
I have experienced that sometimes fixed wing missions result in a flyaway

[![Demo](https://img.youtube.com/vi/EEpRw_IFSNU/0.jpg)](https://youtu.be/EEpRw_IFSNU)

This seems to have somekind of relation ship with where the takeoff waypoint is located related to the vehicle, but this is not consistent. Sometimes the same mission results in a flyaway / or not.

For me, this only happens when using the flightgear SITL simulation

Tested on Ubuntu Bionic, Flightgear 2019.1.1

**Comment by slimonslimon** [comment]
hmmm... Im not able to reproduce this issue after last upstream/master merge.

**Comment by Jaeyoung-Lim** [comment]
@slimonslimon What do you mean by upstream/master merge? 

I have tried the branch on https://github.com/PX4/Firmware/pull/14539, just now, but the problem still persists.

Here is the log: https://review.px4.io/plot_app?log=c26ab6d6-c36e-45d3-8178-d18044b2c748

Actually, for me now it is always the case... there seems to be some interaction with px4 going on. When I do a "goto" the vehicle then follows the waypoint and after that can continue the mission.

**Comment by slimonslimon** [comment]
I tested it about 10 times, but I'm not able to set up a mission, which fails. I will try this in the evening.

Please, look at PX4 console, is there any message like: 
```INFO  [commander] Takeoff detected```

I have much more logged messages in my log https://review.px4.io/plot_app?log=99d5f09b-a9e8-4153-8af6-7d5f753e7e33

**Comment by Jaeyoung-Lim** [comment]
@slimonslimon which environment are you testing in? Could it be something different coming from the system?

**Comment by slimonslimon** [comment]
I forgot to say, that before I saw this situation on my PC..... and @kaklik has the same situation. So I think it should not be something with the system. 

I have Debian 9.x, GCC 6.x, FG 2019.1.1, or Next (aka 2019.2) both from the source. (distribution version is old 2016.x.x, I can test it on this version too) 
Maybe, I use an old version of QGC (3.5.1)

Generally, The PX4 is connected to FG over the bridge, and it is consuming so few information from its property tree, so it should work with most versions of FG. Most problems come from low FPS of FG which limits the frequency of messages from FG.
But FG is full of "easter egg" without any direct documentation, so very often something stop work and then magically start to work. (I spent half hour today by searching why TerraSync stop work, and finally, it starts work with the same set of parameters as I have before)

Now more about this problem, how looks the mission plan? Depend the situation on it?

**Comment by Jaeyoung-Lim** [comment]
@slimonslimon I can share you the mission file, but for me it happens all the time. So I think it maybe that it doesn't depend "only" on the mission.

The FPS for me ranges from 80 - 150. I looked at the flightgear bridge code and this should be sufficient. 

It seems like you and I are running a different flight gear distribution, although the version tags match. since I didn't build it from source. I used the distributed version from 
```
sudo add-apt-repository ppa:saiarcot895/flightgear
sudo apt update
sudo apt install flightgear
```
I also noticed that my Rascal doesn't have a smoke trail. Could this be something related to the model I have installed in my system?

**Comment by Jaeyoung-Lim** [comment]
@slimonslimon Since this is not working with bionic, could you also document on which version combination this has been tested?

**Comment by kaklik** [comment]
> 
> I also noticed that my Rascal doesn't have a smoke trail. Could this be something related to the model I have installed in my system?

There could be several possible differences.  I wrote some basic explanation in [the new documentation](https://github.com/ThunderFly-aerospace/PX4Devguide/blob/flightgear/en/simulation/flightgear_vehicles.md)

**Comment by Jaeyoung-Lim** [comment]
@kaklik I still see this problem time to time, why was this marked as invalid?

**Comment by kaklik** [comment]
I am unable to reproduce this behavior using the up to date code.

**Comment by slimonslimon** [comment]
there are lots of updates in PX4 and this bridge, is this problem still reproducable? Last update of this bridge stabilized gyroscope. So It is harder (or may not possible) to "confuse QGC horizont and compass" i.e. confuse PX4 position estimator.

**Comment by Jaeyoung-Lim** [comment]
@slimonslimon Yes, this problem comes from the fact that the flightgear bridge is not on lockstep. Therefore if you have a laptop that is not powerful enough, you will be able to reproduce this problem.
